app/code/Magento/Analytics/ReportXml/QueryFactory.php

```diff
@@ -101,12 +113,29 @@ private function constructQuery($queryName)
             $selectBuilder = $assembler->assemble($selectBuilder, $queryConfig);
         }
         $select = $selectBuilder->create();
+        return $this->createQueryObject(
+            $select,
+            $selectBuilder->getConnectionName(),
+            $queryConfig
+        );
+    }
+
+    /**
+     * Create query class using objectmanger
+     *
+     * @param  Select $select
+     * @param  string $connection
+     * @param  array  $queryConfig
+     * @return Query
+     */
+    private function createQueryObject($select, $connection, $queryConfig)
```

Comment:
Could you define argument types, return type and move this new method below public methods ?